charms/worker/k8s/src/token_distributor.py

```diff
@@ -147,70 +271,10 @@ def __init__(self, charm: ops.CharmBase, node_name: str, api_manager: K8sdAPIMan
         self.charm = charm
         self.node_name = node_name
         self.api_manager = api_manager
-        self.token_creation_strategies = {
-            TokenStrategy.CLUSTER: self._create_cluster_token,
-            TokenStrategy.COS: self._create_cos_token,
+        self.token_strategies: Dict[TokenStrategy, TokenManager] = {
+            TokenStrategy.CLUSTER: ClusterTokenManager(api_manager),
+            TokenStrategy.COS: CosTokenManager(api_manager),
         }
-        self.token_revoking_strategies = {
-            TokenStrategy.CLUSTER: self._revoke_cluster_token,
-            TokenStrategy.COS: self._revoke_cos_token,
-        }
```

Comment:
Collapse these 2 dicts into 1

This allows the `token_strategies` to be handled now by a `TokenManager` class with common methods and attributes inherited from the base class